packages/noya-designsystem/src/utils/sketchPattern.ts

```diff
@@ -20,12 +19,9 @@ export function getPatternBackground(
   image?: Sketch.FileRef | Sketch.DataRef,
 ) {
   if (!image || !images[image._ref])
-    return `url(
-        'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYAQMAAADaua+7AAAABlBMVEUAAAAAAAClZ7nPAAAAAXRSTlMAQObYZgAAABNJREFUCNdjYOD/TxL+/4GBFAwAvMsj3bQ3H74AAAAASUVORK5CYI=A',
-      )`;
+    return `'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYAQMAAADaua+7AAAABlBMVEUAAAAAAAClZ7nPAAAAAXRSTlMAQObYZgAAABNJREFUCNdjYOD/TxL+/4GBFAwAvMsj3bQ3H74AAAAASUVORK5CYI=A',`;
```

Comment:
Let's import `CHECKERED_BACKGROUND_BYTES` here, so we don't have to handle strings later.

We just need to export it from here: https://github.com/noya-app/noya/blob/ad1cee93813e33f299b2286ec39d4e979104e83b/packages/noya-renderer/src/hooks/useCheckeredFill.ts#L15

Reply:
I realized there's an issue with this default image approach: we don't put the image data into the `FileMap`, so it won't get rendered by the canvas. We should probably set all that up correctly when we add the `image` object to the fill: 

https://github.com/noya-app/noya/blob/ad1cee93813e33f299b2286ec39d4e979104e83b/packages/noya-state/src/reducers/styleReducer.ts#L327

We may need to handle some portion of this in the `applicationReducer` so that we have access to `sketch.images`

Reply:
It's actually valid to have a pattern with no image - Sketch allows deleting the image. So we may need to handle `undefined` as a return value from this function, or not call it in the first place if there's no `image`

Reply:
wait so I write the placeholder checkered_background in the image on creation? and what should it show when you don't have any _ref ? Just all white?

Reply:
Okey so I should add this CHECKERED_BACKGROUND to the images if it doesn't exist, i think i got it